LDAR_Sim/src/virtual_world/sources.py

```diff
@@ -285,8 +285,14 @@ def generate_emissions(
 
         # Initialize a counter to give all emissions for the source a unique "ID"
         leak_count: int = 0
+        # Pre-calculate date ranges
+        pre_existing_dates = [
+            sim_start_date - timedelta(days=x) for x in range(1, self._emis_duration + 1)
```

Comment:
Is this faster or is date range faster - You do it this way up here and with date range below - can we pick the best one and be consistent?

Reply:
date_range is faster for our use. Updated.